L1Trigger/TrackFindingTracklet/interface/imath.h

```diff
@@ -0,0 +1,1122 @@
+// // Integer representation of floating point arithmetic suitable for FPGA designs
+//
+// Author: Yuri Gershtein
+// Date:   March 2018
+//
+// Functionality:
+//
+//  *note* all integers are assumed to be signed
+//
+//  all variables have units, stored in a map <string,int>, with string a unit (i.e. "phi") and int the power
```

Comment:
This looks like a terribly inefficient approach: keeping track of the units is laudable, but it can be done with much more efficient ways than using a map.

[suggestion]
A similar approach with a smaller runtime cost could be to use an `enum` to list all known units and an `std::array` to keep track of their "exponent".

If the runtime cost is relevant, a better approach could be devised, encoding the units in the actual data types.

Reply:
With units here we mean how the different values are represented as integer in the FPGA. We can not really list these in an enum since it depends on what calculation are done how we will represent them. That said - we should look at how we can improve this code. It is clearly the performance bottleneck. However, performance has not been the goal - we are trying to understand the algorithm and optimizing code has not been a priority.

Reply:
This was left as "unresolved" - answered by @aryd above.